lib/super_digit.rb

```diff
@@ -1,15 +1,23 @@
 # Superdigit
 
-# Time Complexity - ?
-# Space Complexity - ?
+# Time Complexity - O(n2)
+# Space Complexity - O(2n)
 def super_digit(n)
-   
+  nums = n.to_s.split('')
+  
+  total = (nums.map { |str| str.to_i }).sum
+  return total if total < 10
+  return super_digit(total)
 end
   
 
-# Time Complexity - ?
-# Space Complexity - ?
+# Time Complexity - O(n2)
+# Space Complexity - O(2n)
 def refined_super_digit(n, k)
-    
+  nums = n.to_s.split('') * k
+  
+  total = (nums.map { |str| str.to_i }).sum
+  return total if total < 10
+  return super_digit(total)
```

Comment:
This works, but you can improve the bigO.

```suggestion
  sum = super_digit(n)
  sum *= k
  return super_digit(sum)
```